packages/react-app/components/TokenContract.tsx

```diff
@@ -0,0 +1,489 @@
+              onClick={() => {
+                closeSnackbar(key);
+              }}
+            >
+              X
+            </Button>
+          </>
+        );
+        enqueueSnackbar("Transaction sent", {
+          variant,
+          action,
+        });
+      });
+    } catch (e) {
+      console.log(e);
+    }
+  };
+
+  const getTokenName = async () => {
+    const tokenName = await contract.methods.name().call();
+    setTokenName(tokenName);
+  };
+
+  const getTokenSymbol = async () => {
+    const tokenSymbol = await contract.methods.symbol().call();
+    setTokenSymbol(tokenSymbol);
+  };
+
+  const getTokenInitialSupply = async () => {
+    const tokenInitialSupply = await contract.methods
+      .initialSupply()
+      .call()
+      .then((result) => {
+        return ethers.utils.formatUnits(result, 18);
+      });
+    setTokenInitialSupply(tokenInitialSupply);
+  };
+
+  const getTokenDecimals = async () => {
+    const tokenDecimals = await contract.methods.decimals().call();
+    setTokenDecimals(tokenDecimals.toString());
+  };
+
+  const getTokenTotal = async () => {
+    const tokenTotal = await contract.methods.totalSupply().call();
+    const convertedTotal = ethers.utils.formatUnits(tokenTotal.toString(), 18);
+    setTokenTotal(convertedTotal);
+  };
+
+  const getBalance = async () => {
+    const tokenBalance = await contract.methods
+      .balance()
+      .call()
+      .then((result) => {
+        return ethers.utils.formatUnits(result, 18);
+      });
+    setTokenBalance(tokenBalance);
+  };
+
+  const getTokenBalanceOf = async (e) => {
+    e.preventDefault();
+    console.log(address);
+    const tokenBalanceOf = await contract.methods
+      .balanceOf(address)
+      .call()
+      .then((result) => {
+        return ethers.utils.formatUnits(result, 18);
+      });
+    setTokenBalanceOf(tokenBalanceOf);
+  };
+
+  const transferTokens = async (e) => {
+    e.preventDefault();
+    let tx = await contract.methods
+      .transfer(newAddress, tokenTransferValue)
+      .call();
+    let receipt = await tx.wait();
+    console.log("receipt", receipt);
+  };
+
+  return (
+    <Grid sx={{ m: 1 }} container justifyContent="center">
+      <Grid item xs={6} sx={{ m: 2 }}>
+        <Typography gutterBottom variant="h5">
+          Token Contract
+        </Typography>
+        <Typography variant="body2" color="text.secondary">
+          Create and manage your own token on Celo.
+        </Typography>
+        {contractData ? (
+          <Link href={contractLink} target="_blank">
+            {truncateAddress(contractData?.address)}
+          </Link>
+        ) : (
+          <Typography variant="body2" color="text.secondary">
+            No contract detected for {network.name}
+          </Typography>
+        )}
+
+        {/* Token Name */}
+        <Box pt={2}>
+          <Card sx={{ minWidth: 275 }}>
+            <CardContent>
+              {/* <HomeIcon style={iconStyle} color="primary" /> */}
```

Comment:
Should these be removed?

Reply:
Might add back later they look nice but will remove for now.